### PR TITLE
Enable bf16 float32 mkldnn matmul when float32 precision is 'medium'

### PR DESCRIPTION
**Pull request by robert-hardwick**
This fixes an issue on AArch64 cpus supporting BF16, caused when torch.set_float32_matmul_precision("highest") does not disable the bf16 downconversion in mkldnn_matmul.

This was discovered from a unit test failure where the decorator `torch.testing._internal.common_mkldnn.bf32_on_and_off`, which internally switches the float32_matmul_precision between "medium" and "highest" was not having the desired effect.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: aten/src/ATen/native/mkldnn/Matmul.cpp [hunk 1]
```diff
@@ -276,7 +276,7 @@ void mkldnn_matmul(
   // but mkldnn matmul primitive only support bias be 1-D tensors
   // to address their differences, we use mkldnn post ops to perform a fused "add" after matrix multiplication is over
   if (beta != 0.0f) op_attr = ideep::attr_t::fuse_sum();
-  if (mat1.scalar_type() == at::kFloat) op_attr.set_fpmath_mode(dnnl_fpmath_mode_bf16); // bf32 path
+  if (mat1.scalar_type() == at::kFloat && use_mkldnn_bf32_matmul()) op_attr.set_fpmath_mode(dnnl_fpmath_mode_bf16); // bf32 path
```
Comment (jgong5) [inline aten/src/ATen/native/mkldnn/Matmul.cpp:279]: Can we do `bf32_usable` too?

**Review by jgong5 — COMMENTED** [review]
@zhuhaozhe can you take a look since it is related to bf32 code changes.

File: aten/src/ATen/native/mkldnn/Matmul.cpp (hunk 1 above)
Comment (robert-hardwick) [reply aten/src/ATen/native/mkldnn/Matmul.cpp]: @jgong5 i've fixed this now

**Review by jgong5 — APPROVED** [review]

**Comment by jgong5** [comment]
> > Can't add following labels to PR: ciflow/linux-aarch64. Please ping one of the reviewers for help.
> 
> @jgong5 is this because I don't have permission to add a label, or this label is restricted? Thanks

Yes, please ping me if you need any help. I triggered the CI now.

**Comment by robert-hardwick** [comment]
The lint fail doesn't look related to my change, i'll resync my fork

**Comment by robert-hardwick** [comment]
> Pull workflow has not been scheduled for the PR yet. It could be because author doesn't have permissions to run those or skip-checks keywords were added to PR/commits, aborting merge. Please get/give approval for the workflows and/or remove skip ci decorators before next merge attempt. If you think this is a mistake, please contact PyTorch Dev Infra.

@jgong5 looks like i don't have permissions to merge, would you mind doing it?

**Pull request closed without merging**